### ISSUE TITLE
Lua error: `bad argument #1 to 'SetValue' (must be a finite number - Usage: self:SetValue(value))

### ISSUE THREAD

**Issue by ryankhart** [Tercioo/Plater-Nameplates#140]
**Game and Plater version**
Which is the game version and Plater version you're using?
Game version: 10.0.0.46597
Plater version: Plater-v538-Retail - DF v390 - 10.0.0

**Describe the bug**
- Steps to Reproduce (what did you do to make the bug happen):
I'm in the Temple of Purity during the attack happens early in Shadowlands. One of the campaign quests for this area is called "Dangerous Discourse".
This lua error happens when I walk to various areas. I haven't quite pinned it down yet.

- Result (what happens when you follow the steps above):
- I get the lua error that you can see below.

- Expected (what you think the expected behavior would be when following the steps):
No lua error

- When the bug started (a date, wow path release, details update, etc)?
I don't have any clue because I just started playing WoW.

- Additional information:

**Lua Error**
```
Message: Interface/AddOns/Plater/Plater.lua:8698: bad argument #1 to 'SetValue' (must be a finite number - Usage: self:SetValue(value))
Time: Sun Nov 13 17:56:17 2022
Count: 20985
Stack: Interface/AddOns/Plater/Plater.lua:8698: bad argument #1 to 'SetValue' (must be a finite number - Usage: self:SetValue(value))
[string "=[C]"]: in function `SetValue'
[string "@Interface/AddOns/Plater/Plater.lua"]:8698: in function `AnimateFunc'
[string "@Interface/AddOns/Plater/Plater.lua"]:6402: in function <Interface/AddOns/Plater/Plater.lua:6093>

Locals: (*temporary) = NamePlate15PlaterUnitFrameHealthBar {
 ExecuteGlowDown = Texture {
 }
 OnTick = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:7114
 isNamePlate = true
 GetTexCoord = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6728
 isCustomShieldHook = true
 incomingHealIndicator = Texture {
 }
 SetTexCoord = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6724
 GetBackdropColor = <function> defined @Interface/SharedXML/Backdrop.lua:390
 actorLevel = FontString {
 }
 SetUnit = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:7023
 unitName = FontString {
 }
 UNIT_MAXHEALTH = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:7235
 currentAbsorb = 0
 ExecuteRangeBar = Texture {
 }
 SetHook = <function> defined @Interface/AddOns/Plater/libs/DF/mixins.lua:368
 barTexture = Texture {
 }
 AnimationEnd = 0
 questProgressTextFrame = <table> {
 }
 SetAlphaTo = <function> defined @Interface/AddOns/Plater/Plater.lua:8490
 SetBorderTexture = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6828
 barTextureMask = MaskTexture {
 }
 executeRange = Texture {
 }
 DetailsRealTime = FontString {
 }
 AnimationStart = Infinite
 unit = "nameplate4"
 shieldIndicator = Texture {
 }
 CreateTextureMask = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6794
 GetBorderColor = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6876
 GetBorderTexture = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6844
 WidgetType = "healthBar"
 R = 0.745098
 G = 0.188235
 B = 0.113725
 GetEdgeSize = <function> defined @Interface/SharedXML/Backdrop.lua:188
 AnimateFunc = <function> defined @Interface/AddOns/Plater/Plater.lua:8691
 A = 1
 AddMaskTexture = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6783
 IsAnimating = true
 CurrentHealthMax = 0
 0 = <userdata>
 OnBackdropLoaded = <function> defined @Interface/SharedXML/Backdrop.lua:152
 HealthBarEvents = <table> {
 }
 GetColor = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6737
 SetBackdropColor = <function> defined @Interface/SharedXML/Backdrop.lua:399
 displayedUnit = "nameplate4"
 SetColor = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6732
 barBorderTextureForMask = Texture {
 }
 ClearBackdrop = <function> defined @Interface/SharedXML/Backdrop.lua:282
 ExecuteRangeHealthCutOff = Texture {
 }
 UNIT_HEAL_PREDICTION = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:7242
 PlateFrame = NamePlate15 {
 }
 ExtraRaidMark = Texture {
 }
 GetMaskTexture = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6748
 GetTexture = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6712
 HookList = <table> {
 }
 FrameOverlay = NamePlate15PlaterUnitFrameHealthBarOverlayFrame {
 }
 Settings = <table> {
 }
 SetMaskAtlas = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:6769
 ExecuteGlowUp = Texture {
 }
 background = Texture {
 }
 lifePercent = FontString {
 }
 SetupTextureCoordinates = <function> defined @Interface/SharedXML/Backdrop.lua:214
 HasBackdropInfo = <function> defined @Interface/SharedXML/Backdrop.lua:278
 dversion = 390
 UpdateHealth = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:7134
 FadeAnimation = AnimationGroup {
 }
 UNIT_HEAL_ABSORB_AMOUNT_CHANGED = <function> defined @Interface/AddOns/Plater/libs/DF/panel.lua:7254
 ApplyBackdrop = <function> d
```

**Screenshots**
Besides the lua error, I haven't yet noticed anything visually wrong.

**Comment by ryankhart** [comment]
Oops, I fumbled my keyboard somehow this got posted before I was done. I'm still working on it.

**Comment by ryankhart** [comment]
I've updated to the latest alpha version to see if I still get the error.

**Comment by ryankhart** [comment]
Well I haven't got the lua error again ever since updating to version "Plater-v538-35-gfeef016-Retail - DF - v390 - 10.0.0"

But I don't know if that is only because I did a /reload and it was a one time fluke or if the this alpha version has something that fixes it.

I'll try reverting to the release version to test.

**Comment by ryankhart** [comment]
So it seems that this issue is fixed in the latest alpha from my testing. After I reverted to the release version, it eventually gave the error again, and then I switched back to the latest alpha, and hasn't happened to me ones while on "Plater-v538-35-gfeef016-Retail - DF - v390 - 10.0.0"

So, I guess this issue could be closed when the next stable release of the addon is available.

**Comment by cont1nuity** [comment]
Was fixed in alpha with daceef67582136e12a6031544172093257e2f860
